lib/mix/lib/mix/release.ex

```diff
@@ -0,0 +1,389 @@
+    ROOTDIR="$(dirname "$(dirname "$BINDIR")")"
+    EMU=beam
+    PROGNAME=$(echo "$0" | sed 's/.*\///')
+    export EMU
+    export ROOTDIR
+    export BINDIR
+    export PROGNAME
+    exec "$BINDIR/erlexec" ${1+"$@"}
+    """)
+
+    File.chmod!(Path.join(destination, "bin/erl"), 0o744)
+    true
+  end
+
+  @doc """
+  Copies the given application specification into the release.
+
+  It assumes the application exists.
+  """
+  @spec copy_app(t, application) :: boolean()
+  def copy_app(release, app_spec) do
+    app = elem(app_spec, 0)
+    vsn = elem(app_spec, 1)
+
+    source_app = Application.app_dir(app)
+    target_app = Path.join([release.path, "lib", "#{app}-#{vsn}"])
+
+    if skip_app?(release, source_app) do
+      false
+    else
+      File.rm_rf!(target_app)
+      File.mkdir_p!(target_app)
+
+      copy_ebin(release, Path.join(source_app, "ebin"), Path.join(target_app, "ebin"))
+
+      for dir <- @copy_app_dirs do
+        source_dir = Path.join(source_app, dir)
+        target_dir = Path.join(target_app, dir)
+        File.exists?(source_dir) && File.cp_r!(source_dir, target_dir)
+      end
+
+      true
+    end
+  end
+
+  defp skip_app?(%{erts_source: nil}, source_app),
+    do: String.starts_with?(source_app, List.to_string(:code.lib_dir()))
+
+  defp skip_app?(_, _), do: false
+
+  @doc """
+  Copies the ebin directory at `source` to `target`
+  respecting release options such a `:strip_beams`.
+  """
+  @spec copy_ebin(t, Path.t(), Path.t()) :: boolean()
+  def copy_ebin(release, source, target) do
+    with {:ok, [_ | _] = files} <- File.ls(source) do
+      File.mkdir_p!(target)
+      strip_beams? = Keyword.get(release.options, :strip_beams, true)
+
+      for file <- files do
+        source_file = Path.join(source, file)
+        target_file = Path.join(target, file)
+
+        with true <- strip_beams? and String.ends_with?(file, ".beam"),
+             {:ok, {_, chunks}} <- read_significant_chunks(File.read!(source_file)) do
+          File.write!(target_file, build_beam(chunks))
+        else
+          _ -> File.copy(source_file, target_file)
+        end
+      end
+
+      true
+    else
+      _ -> false
+    end
+  end
+
+  defp read_significant_chunks(binary) do
+    :beam_lib.chunks(binary, @significant_chunks, [:allow_missing_chunks])
+  end
+
+  defp build_beam(chunks) do
+    chunks = for {name, chunk} <- chunks, is_binary(chunk), do: {name, chunk}
+    {:ok, binary} = :beam_lib.build_module(chunks)
+    {:ok, fd} = :ram_file.open(binary, [:write, :binary])
+    {:ok, _} = :ram_file.compress(fd)
+    {:ok, binary} = :ram_file.get_file(fd)
+    :ok = :ram_file.close(fd)
+    binary
+  end
+
+  @doc """
+  Copies the cookie to the given path.
+
+  If a cookie option was given, we compare it with
+  the contents of the file (if any), and as the user
+  if they want to override.
+
+  If there is no option, we generate a random one
+  the first time.
+
+  Returns a boolean saying if the cookie was copied
```

Comment:
I might be wrong, but I feel it reads better and it's more explicit:

```
  Returns true if the cookie was copied, false otherwise.
```

Reply:
Tks, fixed locally!